File: cpp v2/Instructor.cpp
// File: Instructor.cpp
#pragma once
#include <string>
#include <iostream>
#include "User.cpp" 
using namespace std;

class Instructor : public User {
private:
    string expertise;

public:
    Instructor() {}
    Instructor(string id, string nm, string exp) : User(id, nm), expertise(exp) {}

    ~Instructor() override {}

    string getExpertise() const { return expertise; }
    void setExpertise(string exp) { expertise = exp; }

    void displayInfo() const override {
        cout << "Instructor: " << name << " (ID: " << userId
            << ") | Expertise: " << expertise << endl;
    }
};

File: py/Instructor.py
from User import User
from abc import ABC, abstractmethod
class Instructor(User):
    def __init__(self, userId, name, expertise):
        super().__init__(userId, name)
        self.__expertise = expertise   # private

    def getExpertise(self):
        return self.__expertise

    def displayInfo(self):
        print(f"Instructor: {self._name} ({self._id}) | Expertise: {self.__expertise}")
